### PR TITLE
Opera webdriver manager fixes

### PR DESCRIPTION
**Pull request by ArunaBalasubramanian**

### CODE REVIEW

**Comment by ArunaBalasubramanian** [comment]
Hey @SergeyPirogov I have fixed all the issues with respect to Opera driver. Can you take a look at it please?

NOTE: 6 unit test failures are pertaining to Git API limit

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian this problem with API rate limit is awful. I have an idea to write some proxy service that will upload the driver and cache it. So we will not get these errors

**Comment by SergeyPirogov** [comment]
Created issue #81 for adding cache

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian hello, I've fixed problem with api rate limit. there is new cache implemetation. So this pr need fixes

**Comment by ArunaBalasubramanian** [comment]
@SergeyPirogov Hey can I ask why Edge support is revoked?

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian because of edge website https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/. It hard to parse versions

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian things are changing fast( I've refactored a project. Please look at the existing master and example for firefox. Now DriverManager looks like this

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian tests are read. Please fix. Take a lot at tests for firefox. Seems that opera is close to it

**Comment by ArunaBalasubramanian** [comment]
@SergeyPirogov I get the below error,
```
ff = webdriver.Opera(executable_path=driver_path, options=options)
E   TypeError: __init__() got an unexpected keyword argument 'options'
```
 I can't see what is the version of selenium python package. But it works on my local with the latest version selenium 3.141.0. Can you please check?

**Comment by SergeyPirogov** [comment]
@ArunaBalasubramanian as I see here, test failures connected to some Bad credentials https://travis-ci.org/SergeyPirogov/webdriver_manager/builds/601234729?utm_source=github_status&utm_medium=notification review tests for Opera driver please

File: webdriver_manager/opera.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        self.driver = OperaDriver(name=name,
+                                  version=version,
+                                  os_type=os_type,
+                                  url=url,
+                                  latest_release_url=latest_release_url,
+                                  opera_release_tag=opera_release_tag)
+
+    def install(self):
+        driver_path = self.download_driver(self.driver)
+        os.chmod(driver_path, 0o755)
```
Comment (SergeyPirogov) [inline webdriver_manager/opera.py]: according to CI failure, you should delete sum512 file here

Comment (ArunaBalasubramanian) [reply webdriver_manager/opera.py]: @SergeyPirogov How you think fixing this would solve this? I see it working fine on my windows machine (without the fix)

Comment (ArunaBalasubramanian) [reply webdriver_manager/opera.py]: @SergeyPirogov It actually worked the sha512_sum file

File: webdriver_manager/opera.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                                  latest_release_url=latest_release_url,
+                                  opera_release_tag=opera_release_tag)
+
+    def install(self):
+        driver_path = self.download_driver(self.driver)
+        if os.path.isfile(driver_path):
+            os.chmod(driver_path, 0o755)
+            return driver_path
+        else:
+            file_path = None
```
Comment (SergeyPirogov) [inline webdriver_manager/opera.py]: might be None here is condition is false.

File: webdriver_manager/opera.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            for name in os.listdir(driver_path):
+                abs_path = os.path.join(driver_path, name)
+                if 'opera' in name:
+                    os.chmod(abs_path, 0o755)
+                    file_path = abs_path
+                elif 'sha512_sum' in name:
+                    os.remove(abs_path)
+            return file_path
```
Comment (SergeyPirogov) [inline webdriver_manager/opera.py]: None here os condition os false

File: webdriver_manager/driver_cache.py [hunk 4]
```diff
@@ -57,7 +57,8 @@ def save_driver_to_cache(self, response, driver_name, version, os_type):
         file_path = os.path.join(driver_path, filename)
         write_file(response.content, file_path)
         files = self.__unpack(file_path)
-        return os.path.join(driver_path, files[0])
+        file_path = os.path.join(driver_path, files[0])
```
Comment (SergeyPirogov) [inline webdriver_manager/driver_cache.py]: no need to assign a variable here, just return

Comment (ArunaBalasubramanian) [reply webdriver_manager/driver_cache.py]: fixed

File: webdriver_manager/opera.py (hunk 3 above)
Comment (ArunaBalasubramanian) [reply webdriver_manager/opera.py]: Changed the logic as per comment

File: webdriver_manager/opera.py (hunk 2 above)
Comment (ArunaBalasubramanian) [reply webdriver_manager/opera.py]: Changed the logic as per comment
